### DISCUSSION TITLE
How to add "created_by" and "updated_by" columns to tables

### DISCUSSION THREAD

**Discussion by mbalslow** [supabase/supabase#7180] [Show and tell]
It might not be clearly stated anywhere, but you can actually use the "auth.uid()" in your default values on columns and in DB functions similarly to how you would use them in RLS policies.

With this, you can quite easily setup logic to make "created_by" and "updated_by" columns:

![created_by field (002)](https://user-images.githubusercontent.com/49169557/172143211-5b47ae2f-fc7d-450d-bf12-158d286cb2dc.png)

![updated_by field (002)](https://user-images.githubusercontent.com/49169557/172143229-900f7f6b-65f3-4522-a9cd-b50d82d99659.png)

![set_updated_by solution (002)](https://user-images.githubusercontent.com/49169557/172143244-e58ebb4e-c99a-439e-8832-9745bf484ab8.png)

**Comment by chris-lsn** [comment]
Update: The default value should be set to `auth.uid()` instead of `uid()`

**Comment by MathiasSchjoedt-Bavngaard** [comment]
how do you diffirentiat between created_by and updated by

**Reply by quebrandolink** [reply]
```
-- cria uma função de atualizar os campos (createdBy, updatedBy, createdAt, updatedAt)
CREATE OR REPLACE FUNCTION on_created()
RETURNS TRIGGER AS $$
BEGIN
  NEW."updatedBy" := auth.uid();
  NEW."createdBy" := auth.uid();
  NEW."createdAt" := NOW();
  NEW."updatedAt" := NOW();
  RETURN NEW;
END;
$$ LANGUAGE plpgsql;

-- cria uma função de atualizar os campos (updatedBy, updatedAt)
create or replace function on_updated()
returns trigger as $$
  begin
    NEW."updatedBy" := auth.uid();
    NEW."updatedAt" := NOW();
    return new;
  end;
  $$ language plpgsql;

-- before update ->  only updatedBy
create trigger handle_updated_persons before update on persons
  for each row execute procedure on_updated ();

-- before insert -> alter (createdBy, updatedBy)
create trigger handle_insert_persons before insert on persons
  for each row execute procedure on_created();

```

A diferença está em before insert e before update
